### PR TITLE
backend refactor + docker compose

### PR DESCRIPTION
**Pull request by aimenhamed**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      POSTGRESQL_PASSWORD: password
+      POSTGRESQL_DATABASE: mydb
+    ports:
+      - 8080:8080
+    depends_on:
+      - database
+      - postgres
+
+  postgres:
+    image: postgres
```
Comment (lhjt) [inline docker-compose.yml:58]: Can you specify the major version of the database that you'll be using? 14 and 15 are incompatible, so if someone had a previous version of 14 running and they'll try run 15 now, it'll fail as the versions are incompatible.

**Review by lhjt — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (lhjt) [reply docker-compose.yml:58]: Other than that though, looks fine.

Comment (aimenhamed) [reply docker-compose.yml]: done :)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      POSTGRESQL_PASSWORD: password
+      POSTGRESQL_DATABASE: mydb
+    ports:
+      - 8080:8080
+    depends_on:
+      - database
+      - postgres
+
+  postgres:
+    image: postgres@sha256:1785b8611bb46e7b8776841be5febc322cdb84c3b72c12bfed364379ecdf9b0d
```
Comment (lhjt) [inline docker-compose.yml:58]: Oh sorry, I should have been more clear; try specifying an image tag version that is still pinned to a version, but generally remains compatible throughout updates (e.g. `15.1-alpine`)
```suggestion
    image: postgres:15.1-alpine
```

**Review by lhjt — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (aimenhamed) [reply docker-compose.yml]: oh lmao, no worries :) updated now

**Review by joanna209 — APPROVED** [review]

**Comment by lhjt** [comment]
Is there a reason you decided to go with 14 instead of the latest major version?

**Comment by aimenhamed** [comment]
> Is there a reason you decided to go with 14 instead of the latest major version?

i just used the version that we are using in staging atm since it works

**Review by lhjt — APPROVED** [review]

**Pull request merged**